### ISSUE TITLE
Not able to use the app token with GetStreams endpoint

### ISSUE THREAD

**Issue by jscoys** [nicklaw5/helix#3]
Hi,

Maybe I’m not using it correctly, but I’m getting an app token (I checked), then I pass it to my client and then I’m calling the GetStream endpoint the same way that in the documentation. The weird thing is that I’m getting a rate limit of 30requests per second instead of 120.

**Comment by jscoys** [comment]
Oh and I tested with the same client-id, app token and secret and I got a ratelimit-rate of 120 with postman with the same endpoint:

ratelimit-limit →120
ratelimit-remaining →119

**Comment by nicklaw5** [comment]
Do you mind posting the code you're using so that I can replicate your issue. But don't include your tokens, I have my own.

**Comment by nicklaw5** [comment]
Never mind, I've just replicated the issue. It appears the `Client-Secret` header is not being attached to the request.

**Comment by nicklaw5** [comment]
Actually, this appears to work as expected:

```go
client, err := helix.NewClient(&helix.Options{
	ClientID: "your-client-id",
	UserAccessToken: "your-user-access-token",
})
if err != nil {
	log.Fatal(err)
}

resp, _ := client.GetStreams(&helix.StreamsParams{
	First: 1,
})
fmt.Printf("%+v\n", resp
```

Output:

```txt
&{ResponseCommon:{StatusCode:200 Error: ErrorStatus:0 ErrorMessage: RateLimit:{Limit:120 Remaining:119 Reset:1528279075} StreamsMetadataRateLimit:{Limit:0 Remaining:0} ClipsCreationRateLimit:{Limit:0 Remaining:0}} Data:{Streams:[{ID:28982088016 UserID:87056709 GameID:29595 CommunityIDs:[] Type:live Title:Mineski vs Vici Gaming China Dota2 Supermajor Playoffs Day 3 (Lower bracket) ViewerCount:60554 StartedAt:2018-06-06 08:17:57 +0000 UTC Language:en ThumbnailURL:https://static-cdn.jtvnw.net/previews-ttv/live_user_pgl_dota2-{width}x{height}.jpg}] Pagination:{Cursor:eyJiIjpudWxsLCJhIjp7Ik9mZnNldCI6MX19}}}
```

As you can see from the output the rate limit is 120 with 119 requests remaining.

**Comment by jscoys** [comment]
Hello thx for your answer.

In my use case I just want to use the AppAccessToken with getstreams endpoint, so not the useraccesstoken. Can you try to reproduce it with that context?

**Comment by nicklaw5** [comment]
This appears to be the issue: https://github.com/nicklaw5/helix/blob/master/helix.go#L337-L339

We are only attaching the app access token for the [_Create Entitlement Grants Upload_](https://dev.twitch.tv/docs/api/reference/#create-entitlement-grants-upload-url) endpoint, as that is the only endpoint that specifies its use. The Twitch docs aren't very helpful. I wasn't aware that it is possible to use app access tokens for other requests.

**Comment by jscoys** [comment]
Yeah i know the doc doesn’t mention it’s possible or other endpoint, I spent time to search and understand myself ;-) so can you add the possibility to specify the appAccessToken? I guess that if your have app and user access token passed you take user one, and if you have only the app you use this one?

**Comment by nicklaw5** [comment]
Sure can. Did you want to put together a PR yourself? Otherwise I can probably get to it over the weekend?

**Comment by nicklaw5** [comment]
@jscoys That's been fixed in fc3312c69cffef1fcd67ce87037af8e287024897. Thanks for reporting it.

**Comment by jscoys** [comment]
Hey thx a lot, sorry was planning to do it tonight after work... you are too fast for me ;-)
